lib/write.ml

```diff
@@ -467,34 +467,34 @@ let to_file ?len ?std ?(newline = true) file x =
     close_out_noerr oc;
     raise e
 
-let stream_to_buffer ?std ob st =
-  Stream.iter (to_buffer ?std ob) st
+let seq_to_buffer ?std ob st =
+  Seq.iter (to_buffer ?std ob) st
 
-let stream_to_string ?buf ?(len = 256) ?std st =
+let seq_to_string ?buf ?(len = 256) ?std st =
   let ob =
     match buf with
         None -> Buffer.create len
       | Some ob ->
           Buffer.clear ob;
           ob
   in
-  stream_to_buffer ?std ob st;
+  seq_to_buffer ?std ob st;
   let s = Buffer.contents ob in
   Buffer.clear ob;
   s
 
-let stream_to_channel ?buf ?(len=2096) ?std oc st =
+let seq_to_channel ?buf ?(len=2096) ?std oc st =
   let ob =
     match buf with
         None -> Buffer.create len
       | Some ob -> ob
   in
-  stream_to_buffer ?std ob st
+  seq_to_buffer ?std ob st
```

Comment:
This line is missing a line to actually write the buffer `Buffer.output_buffer oc ob`

Reply:
Yep, I just found the same fix. I think it should be fixed separately from this PR.

Reply:
I think it is okay to add this to this PR, but it is up to you.

Reply:
I think the fix should be the suggestion made by @gasche in the comments, otherwise the `seq` has no real value as there is no memory improvement.